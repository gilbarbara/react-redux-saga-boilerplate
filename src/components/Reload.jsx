import React from 'react';
import styled from 'styled-components';

import { Button, Heading } from 'styled-minimal';

export const StyledReload = styled.div`
  button {
    pointer-events: all;
  }
`;

const Reload = () => (
  <StyledReload>
    <Heading as="h6" mb={3}>
      There's a new version of this app!
    </Heading>
    <Button variant="dark" outline size="sm" onClick={() => window.location.reload()}>
      Reload
    </Button>
  </StyledReload>
);

export default Reload;
