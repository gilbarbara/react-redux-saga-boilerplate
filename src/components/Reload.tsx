import { useCallback } from 'react';
import styled from '@emotion/styled';
import { Button, H6 } from '@gilbarbara/components';

export const ReloadWrapper = styled.div`
  button {
    pointer-events: all;
  }
`;

function Reload() {
  const handleClick = useCallback(() => {
    window.location.reload();
  }, []);

  return (
    <ReloadWrapper data-component-name="Reload">
      <H6 mb="lg">There's a new version of this app!</H6>
      <Button bg="black" invert onClick={handleClick} size="xs">
        Reload
      </Button>
    </ReloadWrapper>
  );
}

export default Reload;
