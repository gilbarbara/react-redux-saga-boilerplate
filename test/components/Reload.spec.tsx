import React from 'react';
import { fireEvent, render, screen } from 'test-utils';

import Reload from '~/components/Reload';

describe('Reload', () => {
  const { location } = window;

  beforeAll(() => {
    // @ts-ignore
    delete window.location;
    window.location = { ...location, reload: vi.fn() };
  });

  afterAll(() => {
    window.location = location;
  });

  it('should render properly', () => {
    render(<Reload />);

    expect(screen.getByTestId('Reload')).toMatchSnapshot();

    fireEvent.click(screen.getByRole('button'));

    expect(window.location.reload).toHaveBeenCalledTimes(1);
  });
});
