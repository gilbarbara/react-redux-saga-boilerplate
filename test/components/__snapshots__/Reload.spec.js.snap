// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reload should render properly 1`] = `
.c2 {
  background-color: #fff;
  border: 1px solid #212121;
  color: #212121;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 0px;
  box-shadow: none;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 14px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1;
  padding: 7px 14px;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
}

.c2:disabled {
  pointer-events: none;
}

.c2:hover {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c2:focus {
  outline-color: #212121;
}

.c2 .c3 {
  margin-left: 5px;
}

.c1 {
  font-size: 16px;
  font-family: inherit;
  font-weight: 700;
  line-height: 1.4;
  margin: 16px 0 0;
  margin-bottom: 16px;
}

.c1:first-child {
  margin-top: 0;
}

.c0 button {
  pointer-events: all;
}

<StyledReload>
  <div
    className="c0"
  >
    <StyledHeading
      as="h6"
      mb={3}
    >
      <h6
        className="c1"
      >
        There's a new version of this app!
      </h6>
    </StyledHeading>
    <StyledButton
      animate={false}
      block={false}
      disabled={false}
      onClick={[Function]}
      outline={true}
      size="sm"
      type="button"
      variant="dark"
    >
      <button
        className="c2"
        disabled={false}
        onClick={[Function]}
        size="sm"
        type="button"
      >
        Reload
      </button>
    </StyledButton>
  </div>
</StyledReload>
`;
